LSPS3/README.md

```diff
@@ -0,0 +1,317 @@
+`eopning_fee_valid_until`.
+
+LSPs:
+
+* SHOULD use a cryptographically-verifiable method of generating
+  `opening_fee_promise`, such as a MAC of some serialization of the
+  `opening_fee` and the `opening_fee_valid_until`, then encoded in
+  hexadecimal or other text encoding.
+* MUST ensure the promise is no longer than 512 bytes.
+* SHOULD avoid characters outside of the printable ASCII range.
+
+Clients:
+
+* MUST abort this flow if `opening_fee_promise` exceeds 512 bytes.
+* MUST NOT parse or interpret this string, and only provide it
+  verbatim later.
+
+> **Rationale** By using a promise that must be remembered by the
+> client, the LSP does not have to remember what it offered via this
+> API, reducing state storage, especially if the client then decides
+> that the `opening_fee` is too high and ultimately decides not to
+> continue this flow.
+> By using a cryptographically-verifiable method, the LSP can ensure
+> that it will only honour `opening_fee`s it actually promised.
+> A size limit is imposed so that LSPs cannot burden clients with
+> unreasonable storage requirements.
+
+`min_payment_size` and `max_payment_size` are the limits of the
+payment size, inclusive.
+These are in millisatoshis.
+The payment size is the amount that the client will receive on the
+JIT Channel.
+
+The client MUST ignore any additional fields it does not recognize.
+
+> **Rationale** This allows future versions of this spec to add new
+> fields to this response, without causing problems for clients that
+> only support the current version.
+
+### 2.  Request JIT Channel
+
+The client constructs a request body for a `POST /lsp/jitchannel`
+request, depending on their need.
+
+Example `POST /lsp/jitchannel` request:
+
+```JSON
+{
+    "version": 1,
+    "opening_fee": "546000",
+    "opening_fee_valid_until": "2023-02-23T08:47:30.511Z",
+    "opening_fee_promise": "abcdefghijklmnopqrstuvwxyz",
+    "payment_size": "42000",
+    "client_node_id": "03d4e028a0d4a90868ec202ab684fb0085779defea9ca7553e06146557631eec20"
+}
+```
+
+`version` is the version of this spec that will be used for this
+interaction.
+
+`opening_fee`, `opening_fee_valid_until`, and `opening_fee_promise`
+are the parameters acquired from the previous step.
+Clients MUST copy them verbatim from some previous
+`GET /lsp/jitchannel` response.
+LSPs SHOULD check that the `opening_fee_promise` does in fact
+prove that it previously promised the specified `opening_fee` and
+`opening_fee_valid_until`.
+LSPs SHOULD check that the `opening_fee_valid_until` is not a
+past datetime.
+
+`payment_size` is the amount that the client wants to receive.
+
+The client MUST ensure that `payment_size` is within the previous
+`min_payment_size` and `max_payment_size` parameters from the LSP.
+The LSP MUST validate that the `payment_size` is within the previous
+`min_payment_size` and `max_payment_size` parameters from the LSP.
+
+The client MAY use a "subtractive" computation to get the
```

Comment:
Why the verb is MAY and not MUST here?

Reply:
The client can also do an "additive" computation, where it wants to receive N millisats and the payer is the one who pays for the channel open by getting an invoice with N + `opening_fee`.

Basically "subtractive" computation is client pays, "additive" is payer pays.  Probably should explicitly list both models and say that the client MAY select one or the other for its own decision.

Reply:
I am not sure that there is a real use case where the payer wants to pays fees because the payee has to "setup his account". It's not the same as the regular transaction fees where both the payer and the payee have "an account" and have to use third parties to transfer the amount.

Reply:
Nevertheless, the option is there and the spec does not disallow it.